# HG changeset patch
# User jwe
# Date 749004084 0
#      Sun Sep 26 00:41:24 1993 +0000
# Node ID 46ac21715e231237ef57086dcfc32196a7f72fc8
# Parent  096ef59709b3864f650c11479f4acc74a9522598
[project @ 1993-09-26 00:41:24 by jwe]
(octave_gets): Set octave_gets_line to NULL after freeing it.

diff --git a/src/input.cc b/src/input.cc
--- a/src/input.cc
+++ b/src/input.cc
@@ -124,17 +124,20 @@ static char *octave_gets_line = (char *)
 /*
  * Use GNU readline to get an input line and store it in the history
  * list.
  */
 char *
 octave_gets (void)
 {
   if (octave_gets_line != NULL)
-    free (octave_gets_line);
+    {
+      free (octave_gets_line);
+      octave_gets_line = (char *) NULL;
+    }
 
   if (interactive || forced_interactive)
     {
       char *ps = (promptflag > 0) ? user_pref.ps1 : user_pref.ps2;
       char *prompt = decode_prompt_string (ps);
 
       if (interactive)
 	{
